Aplib.Core/CircularArray.cs

```diff
@@ -0,0 +1,106 @@
+
+namespace Aplib.Core
+{
+    /// <summary>
+    /// An array that wraps around when it reaches its end.
+    /// Functionally works sort of like a queue with indexing.
+    /// </summary>
+    public class CircularArray<T>
+    {
+
+        /// <summary>
+        /// The length of the array.
+        /// </summary>
+        public int Length { get; private set; }
```

Comment:
To accommodate the above conversations, change this to MaxLength, and add a Length component that indicates the actual length of the ExposedQueue?

Reply:
TODO: fix in later ticket